### PR TITLE
Added Call type and object in python

### PR DESCRIPTION
**Pull request by jigold**
I purposefully did not add the Call type to the documentation. Will do this in later PR.

### CODE REVIEW

File: python/hail/representation/genotype.py [hunk 1]
```diff
@@ -309,3 +309,188 @@ def fraction_reads_ref(self):
...
+        return 'Call(gt=%s)' % self._jrep
+
+    def __eq__(self, other):
+        return self._jrep.equals(other._jrep)
+
+    def __hash__(self):
+        return self._jrep.hashCode()
+
+    def _init_from_java(self, jrep):
+        self._jrep = jrep
```
Comment (tpoterba) [inline python/hail/representation/genotype.py]: self._jcall = ...

File: python/hail/representation/genotype.py [hunk 2]
```diff
@@ -309,3 +309,188 @@ def fraction_reads_ref(self):
...
+    :param call: Genotype hard call
+    :type call: int or None
+    """
+
+    @handle_py4j
+    def __init__(self, call):
+        """Initialize a Call object."""
+
+        self._jrep = scala_object(Env.hail().variant, 'Call').apply(call)
+        self._jcall = scala_object(Env.hail().variant, 'Call')
```
Comment (tpoterba) [inline python/hail/representation/genotype.py]:
```python
jrep = ...
self._init_from_java(jrep)
```

File: python/hail/representation/genotype.py [hunk 3]
```diff
@@ -309,3 +309,188 @@ def fraction_reads_ref(self):
...
+
+
+class Call(object):
+    """
+    An object that represents an individual's call at a genomic locus.
+
+    :param call: Genotype hard call
+    :type call: int or None
+    """
+
```
Comment (tpoterba) [inline python/hail/representation/genotype.py]: add static attribute `_call_object = None`

File: python/hail/representation/genotype.py [hunk 4]
```diff
@@ -309,3 +309,188 @@ def fraction_reads_ref(self):
...
+    @handle_py4j
+    def __init__(self, call):
+        """Initialize a Call object."""
+
+        self._jrep = scala_object(Env.hail().variant, 'Call').apply(call)
```
Comment (tpoterba) [inline python/hail/representation/genotype.py]: check if `Call._call_jobject` is None, if it is None, set it, otherwise, use it

**Comment by jigold** [comment]
@tpoterba back to you

**Pull request merged**